examples/splinterdb_apis_example.c

```diff
@@ -0,0 +1,296 @@
+ */
+void
+configure_splinter_instance(splinterdb_config *splinterdb_cfg,
+                            data_config       *splinter_data_cfg,
+                            const char        *filename,
+                            uint64             dev_size, // in bytes
+                            uint64             cache_size)           // in bytes
+{
+   memset(splinterdb_cfg, 0, sizeof(*splinterdb_cfg));
+   splinterdb_cfg->filename   = filename;
+   splinterdb_cfg->disk_size  = dev_size;
+   splinterdb_cfg->cache_size = cache_size;
+   splinterdb_cfg->data_cfg   = splinter_data_cfg;
+   return;
+}
+
+/*
+ * ---------------------------------------------------------------------------
+ * do_inserts_and_lookups()
+ *
+ * Insert a small number of hard-coded key-value pairs. Perform few lookups.
+ */
+void
+do_inserts_and_lookups(splinterdb *spl_handle)
+{
+   int num_kv_pairs = 3;
+   // clang-format off
+   // Insert a set of key/value pairs
+   kv_pair kv_pairs[] = {
+                            { "k-1"       , "value-1" },
+                            { "key-2"     , "value-2" },
+                            { "key-third" , "3rd key's value: March" },
+                        };
+   // clang-format on
+
+   for (int ictr = 0; ictr < num_kv_pairs; ictr++) {
+      do_insert(spl_handle,
+                kv_pairs[ictr].kv_key,
+                strlen(kv_pairs[ictr].kv_key),
+                kv_pairs[ictr].kv_val,
+                strlen(kv_pairs[ictr].kv_val));
+   }
+
+   // Lookup a few keys that are known to exist
+   for (int ictr = 0; ictr < num_kv_pairs; ictr++) {
+      const char *key_data = kv_pairs[ictr].kv_key;
+      size_t      key_len  = strlen(kv_pairs[ictr].kv_key);
+      do_lookup(spl_handle, key_data, key_len);
+   }
+
+   do_lookup(spl_handle, "key-not-found", sizeof("key-not-found"));
+}
+
+/*
+ * ---------------------------------------------------------------------------
+ * do_insert()
+ *
+ * Insert a new key/value pair to a SplinterDB instance.
+ * ----------------------------------------------------------------------------
+ */
+int
+do_insert(splinterdb  *spl_handle,
+          const char  *key_data,
+          const size_t key_len,
+          const char  *value_data,
+          const size_t value_len)
+{
+   slice key   = slice_create(key_len, key_data);
+   slice value = slice_create(value_len, value_data);
+   int   rc    = splinterdb_insert(spl_handle, key, value);
+   return rc;
+}
+
+/*
+ * ---------------------------------------------------------------------------
+ * do_lookup()
+ *
+ * Lookup a value given a key.
+ * ---------------------------------------------------------------------------
+ */
+int
+do_lookup(splinterdb *spl_handle, const char *key_data, const size_t key_len)
+{
+   splinterdb_lookup_result result;
+   splinterdb_lookup_result_init(spl_handle, &result, 0, NULL);
+
+   slice key = slice_create(key_len, key_data);
```

Comment:
Do we always need to provide the key size? or is there a chance I could lookup without providing this key_size?